### ISSUE TITLE
The wrong data in merge cells after the hidden column and additionally an error occurs.

### ISSUE THREAD

**Issue by pnowak** [handsontable/handsontable#6888]
### Description
When we merge cells and hidden columns we will have the wrong data in the merged cell. Besides this, after opening editor there will be the wrong data. And we can't use the arrow.

Furthermore, the wrong data is in the previous cell, above the merged we have `$6999.95` instead of `01/14/2017`. Only when we open the editor there will be the right data. 

Additionally, an error in the console occurs - `Uncaught Error: Assertion failed: Expecting an unsigned number.
    at assert`

<img width="390" alt="Zrzut ekranu 2020-05-14 o 12 48 59" src="https://user-images.githubusercontent.com/1555809/81926968-2808c680-95e3-11ea-99d3-3ac8b7211c4d.png">

<img width="391" alt="Zrzut ekranu 2020-05-14 o 12 49 12" src="https://user-images.githubusercontent.com/1555809/81926972-2a6b2080-95e3-11ea-92ee-c1c61c608ec5.png">

In `7.4.2` - https://jsfiddle.net/e1wcu6qy/
<img width="398" alt="Zrzut ekranu 2020-05-14 o 12 48 22" src="https://user-images.githubusercontent.com/1555809/81926931-17585080-95e3-11ea-80a9-42ab14de7e78.png">

<img width="391" alt="Zrzut ekranu 2020-05-14 o 12 48 31" src="https://user-images.githubusercontent.com/1555809/81926939-1aebd780-95e3-11ea-8ab2-f9d333ce0508.png">

and `8.0.0-beta.1` works well - https://jsfiddle.net/k7z1c4w8/

<img width="394" alt="Zrzut ekranu 2020-05-14 o 12 48 41" src="https://user-images.githubusercontent.com/1555809/81926947-1fb08b80-95e3-11ea-97ff-4248bf451862.png">

<img width="385" alt="Zrzut ekranu 2020-05-14 o 12 48 48" src="https://user-images.githubusercontent.com/1555809/81926954-22ab7c00-95e3-11ea-85c0-da0f0d070fb3.png">

### Demo

https://jsfiddle.net/d1kvg68L/

### Related issue:
https://github.com/handsontable/handsontable/issues/6872

### Your environment
* Handsontable version: 8.0.0.-beta2-rev5
* Browser Name and version:
* Operating System:

**Comment by wszymanski** [comment]
Probably fixed within #6872.

**Comment by AMBudnik** [comment]
The cell is blank in 8.0.0-beta2-rev11 https://jsfiddle.net/AMBudnik/2quwbd4p/

![Zrzut ekranu 2020-06-4 o 15 09 16](https://user-images.githubusercontent.com/10757813/83760534-5ffcaa00-a675-11ea-82f6-78a78908b854.png)


However, we do not get any console errors.

Opening the cell we get

![Zrzut ekranu 2020-06-4 o 15 10 29](https://user-images.githubusercontent.com/10757813/83760700-989c8380-a675-11ea-8bd1-b07c00161585.png)

This is **not** correct. We expect `12/01/2018`, the value that we see comes from a hidden column on the same row index.

**Comment by pnowak** [comment]
Unfortunately, we have an issue with keyboard navigation. I mean when I try to use `Enter` (open/close cell editor) I can't move to the merged cell.
![Jun-04-2020 15-35-44](https://user-images.githubusercontent.com/1555809/83763620-347bbe80-a679-11ea-8350-768de3e06461.gif)

In version 7.4.2 and beta.1 works as expected.
![Jun-04-2020 15-33-44](https://user-images.githubusercontent.com/1555809/83763656-3fceea00-a679-11ea-9307-ba3856e4e9f2.gif)

**Comment by wszymanski** [comment]
> This is **not** correct. We expect `12/01/2018`, the value that we see comes from a hidden column on the same row index.

It's the same like on the master.

**Comment by AMBudnik** [comment]
@wszymanski you're right, now (rev11) works the same as 7.4.2 https://jsfiddle.net/AMBudnik/obj67k80/
It means that the regression is gone even if the issue is still present. Not sure if we should open a new one or rename this one (+ `regression` label removed).

**Comment by wojciechczerniak** [comment]
@AMBudnik just update this one. No work has gone into it yet

**Comment by AMBudnik** [comment]
Ok, so I've removed the regression as we've reverted the behavior to 7.4.2.

This issue should be fixed in another process, as it has been excluded from the v8 scope.

**Comment by aninde** [comment]
Keyboard navigation still does not work, the content is shown only after opening the editor (it was not regression). In my opinion it is still not fixed. 
https://jsfiddle.net/aninde/fk7g6qa4/
![Kapture 2020-06-15 at 15 05 22](https://user-images.githubusercontent.com/30186818/84660955-e6d54080-af19-11ea-85b6-b3470fc32093.gif)

**Comment by aninde** [comment]
I was wrong, described here bug was too complex. The ability to use <kbd>Enter</kbd> to move on the table is still broken in `v8.0.0-beta2-rev14`. Maybe we split it into separate issues?

**rev14**
![enter](https://user-images.githubusercontent.com/30186818/84662249-dcb44180-af1b-11ea-8ca5-bd915d7ea895.gif)

**7.4.2**
![enter-7 4 2](https://user-images.githubusercontent.com/30186818/84662260-df169b80-af1b-11ea-9b2d-07de4d305a27.gif)

Navigation with <kbd>RIGHT</kbd> arrow is still the same. Reported here https://github.com/handsontable/handsontable/issues/6973
![right arrow](https://user-images.githubusercontent.com/30186818/84662439-1ab16580-af1c-11ea-9f99-7f5871abb52a.gif)
![right arro-7 4 2](https://user-images.githubusercontent.com/30186818/84662450-1c7b2900-af1c-11ea-9b94-7c5e9648b176.gif)

**Comment by wojciechczerniak** [comment]
@aninde You were going to close this issue?

Should it be a regression or something that we've fixed for v8 @AMBudnik ?

**Comment by AMBudnik** [comment]
@pnowak originally reported 3 issues

* Furthermore, the wrong data is in the previous cell, above the merged we have $6999.95 instead of 01/14/2017. Only when we open the editor the right data is in.
* error in the console occurs - Uncaught Error: Assertion failed: Expecting an unsigned number. at assert
*  we couldn't use the arrow

Using the same settings for `8.0.0.-beta2-rev18` https://jsfiddle.net/fvok4810/
* no error in the console
* we can open the dropdown menu
* ' above the merged we have $6999.95 instead of 01/14/2017' fixed as well

I think that we can close this issue as solved. I will report the case that I've mentioned above as a separate issue. There are too many ideas for a single demo to keep it together.

**Comment by aninde** [comment]
Closing as an regression fond and solved during testing `8.0.0-beta2`

**Comment by AMBudnik** [comment]
We'll close it with the release.

**Comment by AMBudnik** [comment]
I'm glad to see this issue fixed :) 

For anyone following this subject. We fixed this issue in version 8.0.0-beta2 which has been rechecked and released as Handsontable 8.0.0.

CDN https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.css
https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.js
NPM npm i handsontable
Release Notes https://github.com/handsontable/handsontable/releases
